lib/ProMotion/table/table.rb

```diff
@@ -172,42 +172,43 @@ def edit_mode?
     end
 
     ########## Cocoa touch methods #################
-    def numberOfSectionsInTableView(table_view)
+    def numberOfSectionsInTableView(_)
       self.promotion_table_data.sections.length
     end
 
     # Number of cells
-    def tableView(table_view, numberOfRowsInSection: section)
+    def tableView(_, numberOfRowsInSection: section)
       self.promotion_table_data.section_length(section)
     end
 
-    def tableView(table_view, titleForHeaderInSection: section)
+    def tableView(_, titleForHeaderInSection: section)
       section = promotion_table_data.section(section)
       section && section[:title]
     end
 
     # Set table_data_index if you want the right hand index column (jumplist)
-    def sectionIndexTitlesForTableView(table_view)
+    def sectionIndexTitlesForTableView(_)
       return if self.promotion_table_data.filtered
       return self.table_data_index if self.respond_to?(:table_data_index)
       nil
     end
 
-    def tableView(table_view, cellForRowAtIndexPath: index_path)
+    def tableView(_, cellForRowAtIndexPath: index_path)
       params = index_path_to_section_index(index_path: index_path)
       data_cell = self.promotion_table_data.cell(section: params[:section], index: params[:index])
       return UITableViewCell.alloc.init unless data_cell
       create_table_cell(data_cell)
     end
 
-    def tableView(table_view, willDisplayCell: table_cell, forRowAtIndexPath: index_path)
+    def tableView(_, willDisplayCell: table_cell, forRowAtIndexPath: index_path)
       data_cell = self.promotion_table_data.cell(index_path: index_path)
       table_cell.send(:will_display) if table_cell.respond_to?(:will_display)
       table_cell.send(:restyle!) if table_cell.respond_to?(:restyle!) # Teacup compatibility
     end
 
-    def tableView(table_view, heightForRowAtIndexPath: index_path)
-      (self.promotion_table_data.cell(index_path: index_path)[:height] || table_view.rowHeight).to_f
+    def tableView(_, heightForRowAtIndexPath: index_path)
+      # mp "table: #{table_view.rowHeight} cell: #{self.promotion_table_data.cell(index_path: index_path)[:height].inspect}"
+      (self.promotion_table_data.cell(index_path: index_path)[:height] || tableView.rowHeight).to_f
```

Comment:
@markrickert The relevant change is that I'm referencing the built-in `tableView` property rather than the passed-in `table_view` var. The search creates another UITableView on top of our table view, which is problematic in that it doesn't contain the settings (`rowHeight` in this case) that we need. I removed the local reference so we aren't tempted to use it.

This will all get re-written anyway when we update for iOS 8's new table searchable stuff. But it does work for now.